beancount/parser/lexer.py

```diff
@@ -151,24 +145,10 @@ def NUMBER(self, number):
         # Note: We don't use D() for efficiency here.
         # The lexer will only yield valid number strings.
         if ',' in number:
-            # Extract the integer part and check the commas match the
-            # locale-aware formatted version. This
-            match = re.match(r"([\d,]*)(\.\d*)?$", number)
-            if not match:
-                # This path is never taken because the lexer will parse a comma
-                # in the fractional part as two NUMBERs with a COMMA token in
-                # between.
-                self.errors.append(
-                    LexerError(self.get_lexer_location(),
-                               "Invalid number format: '{}'".format(number), None))
-            else:
-                int_string, float_string = match.groups()
-                reformatted_number = r"{:,.0f}".format(int(int_string.replace(",", "")))
```

Comment:
This was a way to check that the commas were input as per the user locale.
I'm not attached to it, you version is fine, thought I'd let you know why it was there though.

Reply:
I don't remember now if I changed this willingly to have the parsing independent of the locale or if I didn't understand why it was done like this. I like the fact that the parsing is locale independent but I can revert to the old behavior if you prefer.

Reply:
I also prefer parsing as locale independent - but I fear this might break some clients.
Let's see what happens.